openpype/hosts/flame/plugins/publish/integrate_batch_group.py

```diff
@@ -0,0 +1,290 @@
+import os
+import copy
+from pprint import pformat
+import pyblish
+from openpype.lib import get_workdir
+import openpype.hosts.flame.api as opfapi
+import openpype.pipeline as op_pipeline
+
+
+@pyblish.api.log
+class IntegrateBatchGroup(pyblish.api.InstancePlugin):
+    """Integrate published shot to batch group"""
+
+    order = pyblish.api.IntegratorOrder + 0.45
+    label = "Integrate Batch Groups"
+    hosts = ["flame"]
+    families = ["clip"]
+
+    # settings
+    default_loader = "LoadClip"
+
+    def process(self, instance):
+        add_tasks = instance.data["flameAddTasks"]
+
+        # iterate all tasks from settings
+        for task_data in add_tasks:
+            # exclude batch group
+            if not task_data["create_batch_group"]:
+                continue
+
+            # create or get already created batch group
+            bgroup = self._get_batch_group(instance, task_data)
+
+            # load plate to batch group
+            self.log.info("Loading subset `{}` into batch `{}`".format(
+                instance.data["subset"], bgroup.name.get_value()
+            ))
+            self._load_clip_to_context(instance, bgroup)
+
+    def _load_clip_to_context(self, instance, bgroup):
```

Comment:
To be honest I didn't understand what happens here until I've rewritten it... It's really hard to read and there is so much unnecessary loops all over the place. They are not be needed if you would prepare some helper variables.

This is what I've ended up with. This is not change request, but recommendation (also didn't test it):
```
# get all loaders for host
loaders_by_name = {
    loader.__name__: loader
    for loader in op_pipeline.discover_loader_plugins()
}

# get all published representations
published_representations = instance.data["published_representations"]
repre_docs_by_id = {
    repre_id: repre_info["representation"]
    for repre_id, repre_info in published_representations.items()
}

# get all loadable representations
repres_by_name = {
    repre["name"]: repre for repre in instance.data["representations"]
}

# Collect representation ids that should be loaded
representation_ids = []
# Store loader name which should be used to load by representation id
loader_name_by_repre_id = {}
for repre_id, repre_doc in repre_docs_by_id.items():
    repre_name = repre_doc["name"]
    repre = repres_by_name[repre_name]
    if not repre.get("load_to_batch_group"):
        continue

    representation_ids.append(repre_id)
    loader_name_by_repre_id[repre_id] = repre["batch_group_loader_name"]

repre_contexts = get_repres_contexts(representation_ids)

for repre_id, repre_context in repre_contexts.items():
    repre_name = repre_docs_by_id[repre_id]["name"]
    loader_name = loader_name_by_repre_id.get(repre_id)
    if not loader_name:
        self.log.warning((
            "Loader name is not defined for representation \"{}\". Skipping."
        ).format(repre_name))
        continue

    loader = loaders_by_name.get(loader_name)
    if loader is None:
        self.log.warning((
            "Loader with name \"{}\" was not found."
            " Skipping load of representation \"{}\"."
        ).format(loader_name, repre_name))
        continue

     try:
        load_with_repre_context(
            Loader, repre_context, data={"workdir": self.task_workdir}
        )

    except IncompatibleLoaderError:
        self.log.warning((
            "Check allowed representations for Loader `{}` in settings\n"
            "Representaton context is not compatible with loader `{}`.\n{}"
        ).format(Loader.__name__, Loader.__name__, pformat(repre_context)))

    except Exception:
        self.log.error(
            (
                "Failed to load representaton context with loader `{}`.\n{}"
            ).format(Loader.__name__, pformat(repre_context)),
            exc_info=True
        )
```

Reply:
Thank you for the suggestion and feedback @iLLiCiTiT, yeah the code got really easier for orientation now ;)